addon/lib/metrics.js

```diff
@@ -75,8 +75,8 @@ module.exports = {
 
     this.maybePingTelemetry();
 
-    Events.on(EVENT_SEND_METRIC, this.onExperimentPing);
-    Events.on(EVENT_RECEIVE_VARIANT_DEFS, this.onReceiveVariantDefs);
+    Events.on(EVENT_SEND_METRIC, this.onExperimentPing.bind(this));
+    Events.on(EVENT_RECEIVE_VARIANT_DEFS, this.onReceiveVariantDefs.bind(this));
```

Comment:
You'll need to assign the bound functions somewhere and pass 'em to `on` and `off`